src/LightDFG/services/LightDFG.jl

```diff
@@ -72,107 +35,90 @@ function addVariable!(dfg::LightDFG{<:AbstractParams, V, <:AbstractDFGFactor}, v
     push!(dfg.addHistory, variable.label)
 
-    return true
-end
-
-"""
-    $(SIGNATURES)
-Add a DFGFactor to a DFG.
-"""
-function addFactor!(dfg::LightDFG{<:AbstractParams, V, F}, variables::Vector{V}, factor::F)::Bool where {V <: AbstractDFGVariable, F <: AbstractDFGFactor}
-    # if haskey(dfg.g.metaindex[:label], factor.label)
-    #     error("Factor '$(factor.label)' already exists in the factor graph")
-    # end
-    #TODO should this be an error
-    if haskey(dfg.g.factors, factor.label)
-        error("Factor '$(factor.label)' already exists in the factor graph")
-    end
-    # for v in variables
-    #     if !(v.label in keys(dfg.g.metaindex[:label]))
-    #         error("Variable '$(v.label)' not found in graph when creating Factor '$(factor.label)'")
-    #     end
-    # end
-
-    variableLabels = map(v->v.label, variables)
-
-    resize!(factor._variableOrderSymbols, length(variableLabels))
-    factor._variableOrderSymbols .= variableLabels
-    # factor._variableOrderSymbols = copy(variableLabels)
-
-    return FactorGraphs.addFactor!(dfg.g, variableLabels, factor)
+    return variable
 end
 
-"""
-    $(SIGNATURES)
-Add a DFGFactor to a DFG.
-"""
-function addFactor!(dfg::LightDFG{<:AbstractParams, <:AbstractDFGVariable, F}, variableLabels::Vector{Symbol}, factor::F)::Bool where F <: AbstractDFGFactor
-    #TODO should this be an error
+#moved to abstract
+# function addFactor!(dfg::LightDFG{<:AbstractParams, V, F}, variables::Vector{<:V}, factor::F)::F where {V <: AbstractDFGVariable, F <: AbstractDFGFactor}
+#
+#     #TODO should this be an error
+#     if haskey(dfg.g.factors, factor.label)
+#         error("Factor '$(factor.label)' already exists in the factor graph")
+#     end
+#     # for v in variables
+#     #     if !(v.label in keys(dfg.g.metaindex[:label]))
+#     #         error("Variable '$(v.label)' not found in graph when creating Factor '$(factor.label)'")
+#     #     end
+#     # end
+#
+#     variableLabels = map(v->v.label, variables)
+#
+#     resize!(factor._variableOrderSymbols, length(variableLabels))
+#     factor._variableOrderSymbols .= variableLabels
+#     # factor._variableOrderSymbols = copy(variableLabels)
+#
+#     @assert FactorGraphs.addFactor!(dfg.g, variableLabels, factor)
+#     return factor
+# end
+#
+# function addFactor!(dfg::LightDFG{<:AbstractParams, <:AbstractDFGVariable, F}, variableLabels::Vector{Symbol}, factor::F)::F where F <: AbstractDFGFactor
+#     #TODO should this be an error
+#     if haskey(dfg.g.factors, factor.label)
+#         error("Factor '$(factor.label)' already exists in the factor graph")
+#     end
+#
+#     resize!(factor._variableOrderSymbols, length(variableLabels))
+#     factor._variableOrderSymbols .= variableLabels
+#
+#     @assert FactorGraphs.addFactor!(dfg.g, variableLabels, factor)
+#
+#     return factor
+# end
+
+
+function addFactor!(dfg::LightDFG{<:AbstractParams, <:AbstractDFGVariable, F}, factor::F)::F where F <: AbstractDFGFactor
```

Comment:
again, we should not lose the addFactor!(fg, [:x1;:x2], Pose2Pose2) usage so quickly.  We can wrap in IIF is really need to.  Also dont know how I feel about a factor being so specific list of variables if the factor is a standalone thing ... the factor is a function that requires inputs, doesn't have to be that totally specific variable in the factor graph only,

Reply:
We will have to chat about that, if I understand you correctly what is the use of variable order symbols then?

Reply:
also a factor has a unique label tied to variables